src/Nancy/Extensions/ObjectExtensions.cs

```diff
@@ -0,0 +1,27 @@
+namespace Nancy.Extensions
+{
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Dynamic;
+
+    /// <summary>
+    /// Contains extensions to <see cref="object"/> class.
+    /// </summary>
+    public static class ObjectExtensions
+    {
+        /// <summary>
+        /// Convert an object to a dynamic type
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public static dynamic ToDynamic(this object value)
+        {
+            IDictionary<string, object> expando = new ExpandoObject();
+
+            foreach (PropertyDescriptor property in TypeDescriptor.GetProperties(value.GetType()))
+                expando.Add(property.Name, property.GetValue(value));
+
+            return (ExpandoObject)expando;

```

Comment:
Can't be var, has to be dynamic or explicit or he can't modify it. The cast on the return of dynamic assignment wouldn't be needed.